Build failure using `reinterpret_cast` on Windows

Building intel llvm `b87f456ae6389342231e44280bedca3017babe05` with CUDA backend on Windows fails with below error
```
C:\Users\codeplay\llvm\build\_deps\unified-runtime-src\source\adapters\cuda\device.cpp(1170): error C2440: 'reinterpret_cast': cannot convert from 'uintptr_t' to 'ur_native_handle_t'
C:\Users\codeplay\llvm\build\_deps\unified-runtime-src\source\adapters\cuda\device.cpp(1170): note: Conversion is a valid standard conversion, which can be performed implicitly or by use of static_cast, C-style cast or function-style cast
```

--- a/source/adapters/cuda/device.cpp
+++ b/source/adapters/cuda/device.cpp
@@ -1167,8 +1167,7 @@ UR_APIEXPORT ur_result_t UR_APICALL urDeviceGet(ur_platform_handle_t hPlatform,
 
 UR_APIEXPORT ur_result_t UR_APICALL urDeviceGetNativeHandle(
     ur_device_handle_t hDevice, ur_native_handle_t *phNativeHandle) {
-  *phNativeHandle = reinterpret_cast<ur_native_handle_t>(
-      static_cast<std::uintptr_t>(hDevice->get()));
+  *phNativeHandle = static_cast<ur_native_handle_t>(hDevice->get());
   return UR_RESULT_SUCCESS;
 }
 

--- a/test/adapters/cuda/urDeviceCreateWithNativeHandle.cpp
+++ b/test/adapters/cuda/urDeviceCreateWithNativeHandle.cpp
@@ -15,8 +15,7 @@ TEST_F(urCudaDeviceCreateWithNativeHandle, Success) {
     CUdevice cudaDevice;
     ASSERT_SUCCESS_CUDA(cuDeviceGet(&cudaDevice, 0));
 
-    ur_native_handle_t nativeCuda = reinterpret_cast<ur_native_handle_t>(
-        static_cast<std::uintptr_t>(cudaDevice));
+    ur_native_handle_t nativeCuda = static_cast<ur_native_handle_t>(cudaDevice);
     ur_device_handle_t urDevice;
     ASSERT_SUCCESS(urDeviceCreateWithNativeHandle(nativeCuda, platform, nullptr,
                                                   &urDevice));

